src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -162,6 +165,12 @@ def enable_autocomplete(self):
         argcomplete.autocomplete(self, validator=lambda c, p: c.lower().startswith(p.lower()),
                                  default_completer=lambda _: ())
 
+    def _get_values(self, action, arg_strings):
+        value = super(AzCliCommandParser, self)._get_values(action, arg_strings)
+        if action.dest and isinstance(action.dest, str) and not action.dest.startswith('_'):
+            self.specified_arguments.append(action.dest)
```

Comment:
looks tricky to add this logic in a _get_value method

Reply:
this is a function which is called in the `ArgumentParser.parse_args()`. The input for `parse_args()` is the command string(such as `vm create -g rg -n vm_name`), and `_get_values()` is called per parameter. For the given example, it will call `_get_values` for  `-g` and `-n`. And here we want to collect the user specified arguments, I think it is a good place to add logic here.